app.py

```diff
@@ -172,25 +172,22 @@ def results(page_num):
     if 'id_list' in session:
         id_list = session['id_list'] 
     else:
-        id_list = None
-
-    total_pages = 0
-    if id_list:
-        total_pages = math.ceil(len(id_list)/RESULTS_PER_PAGE) - 1
-        
+        return render_template('search_results.html', results=None, total_pages=0, cur_page=0)
```

Comment:
Not sure I understand why user is being directed to (an empty?) search results page here. Wouldn't it be more consistent to use the 404 page, similar to below?

Reply:
No, the difference here is if someone searches for an ingredient, and there are no matches, it just returns that there are no results.